### ISSUE TITLE
Fasttext yields same wrong prediction probabilities on VM with NVIDIA RAPIDS environment

### ISSUE THREAD

**Issue by lvxhnat** [facebookresearch/fastText#1299]
I am currently running fasttext on VertexAI Workbench with the following specifications: 
Environment : RAPIDS 0.18 (with Intel® MKL-DNN/MKL)
Environment version: M65
Machine type : n1-standard-4 (4 vCPUs, 15 GB RAM)
GPU : NVIDIA Tesla T4 x 1

What I realised is that running fasttext on this VM environment will always yield the same probability, screenshot below

```
import fasttext

language_model = fasttext.load_model("assets/models/langdetects/lid.176.bin")

language_model.predict("안녕 잘 지내")
```
The model is installed from https://fasttext.cc/docs/en/language-identification.html. These are the pip dependencies i have in my environment on the VM: 
<details>
aiohttp==3.8.3<br/>
aiosignal==1.2.0<br/>
amqp==5.1.1<br/>
anyio==3.6.1<br/>
appdirs==1.4.4<br/>
argon2-cffi==21.3.0<br/>
argon2-cffi-bindings==21.2.0<br/>
asttokens==2.0.8<br/>
async-timeout==4.0.2<br/>
asyncpg==0.26.0<br/>
attrs==22.1.0<br/>
backcall==0.2.0<br/>
backports.functools-lru-cache==1.6.4<br/>
beautifulsoup4==4.11.1<br/>
billiard==3.6.4.0<br/>
bleach==5.0.1<br/>
blis==0.7.8<br/>
bokeh==2.4.3<br/>
brotlipy==0.7.0<br/>
cachetools==5.2.0<br/>
catalogue==2.0.8<br/>
celery==5.2.7<br/>
certifi==2022.9.24<br/>
cffi==1.15.1<br/>
charset-normalizer==2.1.1<br/>
click==8.0.4<br/>
click-didyoumean==0.3.0<br/>
click-plugins==1.1.1<br/>
click-repl==0.2.0<br/>
cligj==0.7.2<br/>
cloudpickle==2.2.0<br/>
colorama==0.4.5<br/>
colorcet==3.0.0<br/>
confection==0.0.2<br/>
confluent-kafka==1.7.0<br/>
contourpy==1.0.5<br/>
cryptography==37.0.4<br/>
cucim==22.6.0<br/>
cuda-python==11.7.0<br/>
cudf==22.6.1<br/>
cudf-kafka==22.6.1<br/>
cugraph==22.6.1+0.gde8036b5.dirty<br/>
cuml==22.6.1<br/>
cupy==9.6.0<br/>
cupy-cuda113==10.6.0<br/>
cusignal==22.6.0<br/>
cuspatial==22.6.0<br/>
custreamz==22.6.1<br/>
cuxfilter==22.6.0<br/>
cycler==0.11.0<br/>
cymem==2.0.6<br/>
cytoolz==0.12.0<br/>
dask==2022.5.2<br/>
dask-cuda==22.6.0<br/>
dask-cudf==22.6.1<br/>
datashader==0.13.1a0<br/>
datashape==0.5.4<br/>
db-dtypes==1.0.4<br/>
debugpy==1.6.3<br/>
decorator==5.1.1<br/>
defusedxml==0.7.1<br/>
Deprecated==1.2.13<br/>
distributed==2022.5.2<br/>
elastic-transport==8.4.0<br/>
elasticsearch==8.4.0<br/>
emoji==1.7.0<br/>
entrypoints==0.4<br/>
executing==1.1.0<br/>
fastavro==1.6.1<br/>
fastjsonschema==2.16.2<br/>
fastrlock==0.8<br/>
fasttext==0.9.2<br/>
filelock==3.8.0<br/>
Fiona==1.8.20<br/>
flit_core==3.7.1<br/>
fonttools==4.37.3<br/>
frozenlist==1.3.1<br/>
fsspec==2022.8.2<br/>
gcsfs==2022.8.2<br/>
GDAL==3.3.2<br/>
geopandas==0.9.0<br/>
google-api-core==2.8.2<br/>
google-auth==2.11.0<br/>
google-auth-oauthlib==0.5.2<br/>
google-cloud-bigquery==3.3.2<br/>
google-cloud-bigquery-storage==2.14.2<br/>
google-cloud-core==2.3.2<br/>
google-cloud-storage==2.5.0<br/>
google-crc32c==1.5.0<br/>
google-resumable-media==2.4.0<br/>
googleapis-common-protos==1.56.4<br/>
greenlet==1.1.3<br/>
grpcio==1.49.1<br/>
grpcio-status==1.49.1<br/>
HeapDict==1.0.1<br/>
holoviews==1.14.6<br/>
huggingface-hub==0.10.0<br/>
idna==3.4<br/>
imagecodecs==2021.8.26<br/>
imageio==2.22.0<br/>
importlib-metadata==4.11.4<br/>
importlib-resources==5.9.0<br/>
ipykernel==6.14.0<br/>
ipython==8.4.0<br/>
ipywidgets==8.0.2<br/>
jedi==0.18.1<br/>
Jinja2==3.1.2<br/>
joblib==1.2.0<br/>
jsonschema==4.16.0<br/>
jupyter-client==7.3.4<br/>
jupyter_core==4.11.1<br/>
jupyter-server==1.19.1<br/>
jupyter-server-proxy==3.2.2<br/>
jupyterlab-pygments==0.2.2<br/>
jupyterlab-widgets==3.0.3<br/>
kiwisolver==1.4.4<br/>
kombu==5.2.4<br/>
langcodes==3.3.0<br/>
llvmlite==0.39.1<br/>
locket==1.0.0<br/>
lxml==4.8.0<br/>
lz4==4.0.0<br/>
mapclassify==2.4.3<br/>
Markdown==3.4.1<br/>
MarkupSafe==2.1.1<br/>
matplotlib==3.6.0<br/>
matplotlib-inline==0.1.6<br/>
mistune==2.0.4<br/>
modelx==0.1<br/>
more-itertools==8.14.0<br/>
msgpack==1.0.4<br/>
multi-rake==0.0.2<br/>
multidict==6.0.2<br/>
multipledispatch==0.6.0<br/>
munch==2.5.0<br/>
munkres==1.1.4<br/>
murmurhash==1.0.8<br/>
nbclient==0.6.8<br/>
nbconvert==7.0.0<br/>
nbformat==5.6.1<br/>
nest-asyncio==1.5.5<br/>
networkx==2.6.3<br/>
nltk==3.7<br/>
numba==0.56.2<br/>
numpy==1.21.6<br/>
nvtx==0.2.3<br/>
oauthlib==3.2.1<br/>
packaging==21.3<br/>
pandas==1.4.4<br/>
pandas-gbq==0.17.4<br/>
pandocfilters==1.5.0<br/>
panel==0.12.7<br/>
param==1.12.2<br/>
parso==0.8.3<br/>
partd==1.3.0<br/>
pathy==0.6.2<br/>
pexpect==4.8.0<br/>
pickleshare==0.7.5<br/>
Pillow==9.1.1<br/>
pip==22.2.2<br/>
pkgutil_resolve_name==1.3.10<br/>
preshed==3.0.7<br/>
prometheus-client==0.14.1<br/>
prompt-toolkit==3.0.31<br/>
proto-plus==1.22.1<br/>
protobuf==4.21.7<br/>
psutil==5.9.2<br/>
psycopg2-binary==2.9.3<br/>
ptxcompiler==0.2.0<br/>
ptyprocess==0.7.0<br/>
pure-eval==0.2.2<br/>
pyarrow==7.0.0<br/>
pyasn1==0.4.8<br/>
pyasn1-modules==0.2.8<br/>
pybind11==2.10.0<br/>
pycld2==0.41<br/>
pycparser==2.21<br/>
pyct==0.4.6<br/>
pydantic==1.9.2<br/>
pydata-google-auth==1.4.0<br/>
pydeck==0.5.0<br/>
pyee==8.1.0<br/>
Pygments==2.13.0<br/>
pylibcugraph==22.6.1+0.gde8036b5.dirty<br/>
pynvml==11.4.1<br/>
pyOpenSSL==22.0.0<br/>
pyparsing==3.0.9<br/>
pyppeteer==1.0.2<br/>
pyproj==3.1.0<br/>
pyrsistent==0.18.1<br/>
PySocks==1.7.1<br/>
python-dateutil==2.8.2<br/>
python-dotenv==0.20.0<br/>
pytz==2022.2.1<br/>
pyviz-comms==2.2.1<br/>
PyWavelets==1.3.0<br/>
PyYAML==6.0<br/>
pyzmq==24.0.1<br/>
raft==22.6.0<br/>
redis==4.3.4<br/>
regex==2022.9.13<br/>
requests==2.28.1<br/>
requests-oauthlib==1.3.1<br/>
rmm==21.12.0<br/>
rsa==4.9<br/>
Rtree==1.0.0<br/>
scikit-image==0.19.3<br/>
scikit-learn==1.1.2<br/>
scipy==1.9.1<br/>
Send2Trash==1.8.0<br/>
sentence-transformers==2.2.2<br/>
sentencepiece==0.1.97<br/>
setuptools==60.10.0<br/>
Shapely==1.8.0<br/>
simpervisor==0.4<br/>
six==1.16.0<br/>
smart-open==5.2.1<br/>
sniffio==1.3.0<br/>
sortedcontainers==2.4.0<br/>
soupsieve==2.3.2.post1<br/>
spacy==3.4.1<br/>
spacy-legacy==3.0.10<br/>
spacy-loggers==1.0.3<br/>
SQLAlchemy==1.4.40<br/>
srsly==2.4.4<br/>
stack-data==0.5.1<br/>
streamz==0.6.4<br/>
tblib==1.7.0<br/>
terminado==0.15.0<br/>
thinc==8.1.2<br/>
threadpoolctl==3.1.0<br/>
tifffile==2021.11.2<br/>
tinycss2==1.1.1<br/>
tokenizers==0.12.1<br/>
toolz==0.12.0<br/>
torch==1.12.1<br/>
torchvision==0.13.1<br/>
tornado==6.1<br/>
tqdm==4.64.0<br/>
traitlets==5.4.0<br/>
transformers==4.22.2<br/>
treelite==2.4.0<br/>
treelite-runtime==2.4.0<br/>
typer==0.4.2<br/>
typing_extensions==4.3.0<br/>
ucx-py==0.26.0<br/>
unicodedata2==14.0.0<br/>
urllib3==1.26.11<br/>
vine==5.0.0<br/>
wasabi==0.10.1<br/>
wcwidth==0.2.5<br/>
webencodings==0.5.1<br/>
websocket-client==1.4.1<br/>
websockets==10.3<br/>
wheel==0.37.1<br/>
widgetsnbextension==4.0.3<br/>
wrapt==1.14.1<br/>
xarray==2022.6.0<br/>
xgboost==1.6.1<br/>
yarl==1.7.2<br/>
zict==2.2.0<br/>
zipp==3.8.1<br/>
</details>

<img width="987" alt="Screenshot 2022-10-09 at 12 33 42 PM" src="https://user-images.githubusercontent.com/58301316/194738226-e819342d-830e-4bd2-ada1-280805170519.png">

**Comment by keys-zlc** [comment]
Have you ever solved the problem?

**Comment by lvxhnat** [comment]
> Have you ever solved the problem?

It resolved itself, but I never figured out why, or how. My guess is it had something to do with the way the model is cached when calling ```load_model```, since it resolved after resetting my VM.
